File: keevo/app/lib/features/auth/domain/model/login_session_response.dart
import 'membership_dto.dart';

/// LoginSessionResponse — intermediate parse result from POST /auth/login.
///
/// Contains a short-lived loginToken (5 min) and the user's tenant memberships.
/// The datasource returns this; [LoginUseCase] decides whether to auto-select
/// (single membership) or surface the picker (multiple memberships).
class LoginSessionResponse {
  final String loginToken;
  final List<MembershipDto> memberships;

  const LoginSessionResponse({
    required this.loginToken,
    required this.memberships,
  });
}

File: keevo/backend/src/main/java/com/keevo/identity/auth/adapter/in/rest/dto/LoginSessionResponse.java
package com.keevo.identity.auth.adapter.in.rest.dto;

import java.util.List;

/**
 * LoginSessionResponse — REST DTO for POST /auth/login step 1 (Story 1.7).
 *
 * <p>Contains:
 * <ul>
 *   <li>{@code loginToken} — short-lived RS256 JWT (5min, scope=login_pending)</li>
 *   <li>{@code memberships} — list of tenants the user belongs to</li>
 * </ul>
 *
 * <p>Flutter behaviour:
 * <ul>
 *   <li>If {@code memberships.size() == 1} → auto-calls POST /auth/select-tenant (transparent)</li>
 *   <li>If {@code memberships.size() > 1} → shows TenantPickerScreen</li>
 * </ul>
 */
public record LoginSessionResponse(
        String loginToken,
        List<MembershipDto> memberships
) {
    /**
     * MembershipDto — projection of one tenant membership for the login response.
     *
     * @param tenantCode  public tenant code (e.g., "KV-ABC123")
     * @param tenantName  human-readable name (e.g., "Boutique Simon")
     * @param role        user role in this tenant
     * @param schemaName  internal schema name (e.g., "kv_abc123")
     */
    public record MembershipDto(
            String tenantCode,
            String tenantName,
            String role,
            String schemaName
    ) {}
}
